appcues/src/main/java/com/appcues/push/PushOpenedProcessor.kt

```diff
@@ -0,0 +1,56 @@
+package com.appcues.push
+
+import android.content.ActivityNotFoundException
+import android.net.Uri
+import com.appcues.Appcues
+import com.appcues.analytics.AnalyticsTracker
+import com.appcues.data.model.ExperienceTrigger.DeepLink
+import com.appcues.di.component.AppcuesComponent
+import com.appcues.di.component.inject
+import com.appcues.di.scope.AppcuesScope
+import com.appcues.di.scope.inject
+import com.appcues.logging.Logcues
+import com.appcues.ui.ExperienceRenderer
+import com.appcues.util.LinkOpener
+
+internal class PushOpenedProcessor(
+    override val scope: AppcuesScope,
+) : AppcuesComponent {
+
+    private val analyticsTracker by inject<AnalyticsTracker>()
+    private val experienceRenderer by scope.inject<ExperienceRenderer>()
+    private val linkOpener by scope.inject<LinkOpener>()
+    private val appcues by scope.inject<Appcues>()
+    private val logcues by scope.inject<Logcues>()
+
+    private val deferredAction: MutableList<PushOpenedAction> = arrayListOf()
+
+    fun defer(action: PushOpenedAction) {
+        deferredAction.add(action)
+    }
+
+    suspend fun processDeferred(userId: String) {
```

Comment:
processDeferred will run the first item found (single item contains all information needed (properties, deeplink and experienceId)

and will clear the entire list to ensure we are not stacking old actions from other users here.